### PR TITLE
python3Packages.django-compressor: Enable more tests

### PR DESCRIPTION
**Pull request by onny**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vojta001** [comment]
Result of `nixpkgs-review pr 186298` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>healthchecks</li>
    <li>mailman-web</li>
    <li>python310Packages.calmjs</li>
    <li>python310Packages.calmjs-types</li>
    <li>python310Packages.django-compressor</li>
    <li>python310Packages.django-mailman3</li>
    <li>python310Packages.django-sekizai</li>
    <li>python39Packages.calmjs</li>
    <li>python39Packages.calmjs-types</li>
    <li>python39Packages.django-compressor</li>
    <li>python39Packages.django-mailman3</li>
    <li>python39Packages.django-sekizai</li>
    <li>seahub</li>
  </ul>
</details>

**Comment by metatoaster** [comment]
Commenting out the tests prematurely actually hid a very critical problem - that `calmjs.parse` might also not have been imported correctly through the packages set up by Nix.  As I noted in calmjs/calmjs#63 packaging `calmjs` (or any other Python packages) on NixOS should ensure that they be installed correctly first before testing, or alternatively the `PYTHONPATH` _must_ be set up correctly such that the raw sources be tested, and that don't blanket kill all the tests.  The reason this is complicated is because Python's namespace/package is used for `calmjs`, and OpenSUSE at one point produced broken packages which upset downstream users.  Please make sure the minimum set of test passed before this is merged.

File: pkgs/development/python-modules/calmjs/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  propagatedBuildInputs = [
+    calmjs-parse
+    calmjs-types
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/calmjs/default.nix:27]: Can you try the following:

```
preCheck = ''
  rm -rf src/calmjs
'';
```

**Comment by SuperSandro2000** [comment]
> that `calmjs.parse` might also not have been imported correctly through the packages set up by Nix.

I think the source just shadows the import while testing.



> Please make sure the minimum set of test passed before this is merged.

Don't worry, I think we'll manage this :)

---

@onny Can you please try to debug more on your own or seek help in the python matrix channel before opening upstream issues? Many problems are related to how nixpkgs does things or the absend of pinning we do and we don't want to take upstreams time with that.

File: pkgs/development/python-modules/calmjs/default.nix (hunk 1 above)
Comment (onny) [reply pkgs/development/python-modules/calmjs/default.nix:32]: After that pytestCheckHook ran 0 tests :(

**Comment by onny** [comment]
@SuperSandro2000 For the calmjs dependency I know tried to ran
```
  checkPhase = ''
    ${python.interpreter} -m unittest calmjs.tests.make_suite
  '';
```
At least I get
```
Ran 669 tests in 7.495s

FAILED (failures=8, skipped=1)
``` 
The 8 failing tests might depend on `npm` and `yarn`. They might not work because they require internet access too? 

I thought I could try the new `unittestCheckHook` because of the "unittest" referenced in the upstream documentation https://github.com/calmjs/calmjs#testing-the-installation but it will also run 0 tests.

**Comment by metatoaster** [comment]
> They might not work because they require internet access too?

None of the tests require internet access.  Which tests have failed? It is very difficult to run tests on `calmjs` on the raw source before it is registered as part of the working set with the Python installation (it needs to be fully installed as considered by Python), though given distributions for some reason enforce the requirement to run the tests _before_ installation (despite Python packaging convention where tests are only usable after installation given that's the point when imports work normally - yes while I am aware this can be forced with `PYTHONPATH` but this is not standard behavior).

Also, tests that require the package to be fully installed will simply not work (as these tests are against the integration of the fully installed version of this package with the Python packaging system), please set the environment variable `CALMJS_SKIP_INTEGRATION` to a non falsey value (e.g. `1`) and run the test suite against the non-installed version of the source tree.

File: pkgs/development/python-modules/calmjs/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/calmjs/default.nix:32]: Then I am out of ideas.

File: pkgs/development/python-modules/calmjs/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  checkInputs = [
+    nodejs
+    yarn
+  ];
+
+  #pytestFlagsArray = [ "calmjs.tests.make_suite" ];
+
+  checkPhase = ''
+    ${python.interpreter} -m unittest calmjs.tests.make_suite
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/calmjs/default.nix:37]:
```suggestion
  checkInputs = [
    nodejs
    yarn
  ];

  checkPhase = ''
    ${python.interpreter} -m unittest calmjs.tests.make_suite
  '';
```

**Comment by onny** [comment]
I think this is good to go now since we managed to get most of the tests running for django-compressor and disabled some for the new package calmjs for now :)

File: pkgs/development/python-modules/calmjs/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/calmjs/default.nix:27]: you could try to set PYTHONPATH to $out/${python.sitePackages} but we can also continue to try this later

**Pull request merged**